### PR TITLE
Test for OpenVINO Examples

### PR DESCRIPTION
**Pull request by NoushNabi**
Added test for all OpenVINO examples except SD.

### CODE REVIEW

File: .github/workflows/test_openvino_examples.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        pip install -r examples/openvino/question-answering/requirements.txt
+        pip install -r examples/openvino/text-classification/requirements.txt
+
+    - run: free -h
+    - run: lscpu
+    - run: pip freeze
+
+    - name: Test examples
+      run: |
+        python examples/openvino/audio-classification/run_audio_classification.py  --model_name_or_path facebook/wav2vec2-base  --nncf_compression_config examples/openvino/audio-classification/configs/wav2vec2-base-qat.json  --dataset_name superb  --dataset_config_name ks  --max_train_samples 10  --max_eval_samples 2  --output_dir /tmp/qat-wav2vec2-base-ft-keyword-spotting  --overwrite_output_dir  --remove_unused_columns False  --do_train  --learning_rate 3e-5  --max_length_seconds 1  --attention_mask False  --warmup_ratio 0.1  --num_train_epochs 1  --gradient_accumulation_steps 1  --dataloader_num_workers 1  --logging_strategy steps  --logging_steps 1  --evaluation_strategy epoch  --save_strategy epoch  --load_best_model_at_end False  --seed 42
```
[the remark this answers is not in the document]
Comment (NoushNabi) [reply .github/workflows/test_openvino_examples.yml:53]: Sure! I will.

File: .github/workflows/test_openvino_examples.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        python examples/openvino/question-answering/run_qa.py  --model_name_or_path sshleifer/tiny-distilbert-base-cased-distilled-squad  --dataset_name squad  --do_train  --do_eval  --max_train_samples 10  --max_eval_samples 2  --learning_rate 3e-5  --num_train_epochs 1  --max_seq_length 384  --doc_stride 128  --output_dir /tmp/outputs_squad/  --overwrite_output_dir
+        python examples/openvino/image-classification/run_image_classification.py  --model_name_or_path nateraw/vit-base-beans  --dataset_name beans  --max_train_samples 10  --max_eval_samples 2  --remove_unused_columns False  --do_train  --do_eval  --learning_rate 2e-5  --num_train_epochs 1  --logging_strategy steps  --logging_steps 1  --evaluation_strategy epoch  --save_strategy epoch  --save_total_limit 1  --seed 1337  --output_dir /tmp/beans_outputs/
+
```
Comment (echarlaix) [inline .github/workflows/test_openvino_examples.yml:51]:
```suggestion
        python examples/openvino/test_ov_examples.py
```

File: examples/openvino/test_ov_examples.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import os
+import sys
+import tempfile
+import unittest
+from unittest.mock import patch
+
+#os.environ["CUDA_VISIBLE_DEVICES"] = ""
+
```
Comment (echarlaix) [inline examples/openvino/test_ov_examples.py:21]: very minor but would rename it to `test_examples.py` (feel free to close)

File: examples/openvino/test_ov_examples.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    import run_glue
+    import run_qa
+
+
+class TestExamples(unittest.TestCase):
+    def test_audio_classification(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            test_args = f"""
+                run_audio_classification.py  
+                --model_name_or_path facebook/wav2vec2-base
```
Comment (echarlaix) [inline examples/openvino/test_ov_examples.py:45]: I would prefer we replace it with a tiny model :
```suggestion
                --model_name_or_path hf-internal-testing/tiny-random-Wav2Vec2Model
```

File: examples/openvino/test_ov_examples.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                """.split()
+
+            with patch.object(sys, "argv", test_args):
+                run_audio_classification.main()
+
+    def test_image_classification(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            test_args = f"""
+                run_image_classification.py  
+                --model_name_or_path nateraw/vit-base-beans
```
Comment (echarlaix) [inline examples/openvino/test_ov_examples.py:77]:
```suggestion
                --model_name_or_path hf-internal-testing/tiny-random-vit
```

File: examples/openvino/test_ov_examples.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                """.split()
+
+            with patch.object(sys, "argv", test_args):
+                run_image_classification.main()
+
+    def test_text_classification(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            test_args = f"""  
+                run_glue.py  
+                --model_name_or_path sshleifer/tiny-distilbert-base-cased-distilled-squad
```
Comment (echarlaix) [inline examples/openvino/test_ov_examples.py:102]:
```suggestion
                --model_name_or_path hf-internal-testing/tiny-random-DistilBertForSequenceClassification
```

File: examples/openvino/test_ov_examples.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                """.split()
+
+            with patch.object(sys, "argv", test_args):
+                run_glue.main()
+
+    def test_question_answering(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            test_args = f"""
+                run_qa.py  
+                --model_name_or_path sshleifer/tiny-distilbert-base-cased-distilled-squad
```
Comment (echarlaix) [inline examples/openvino/test_ov_examples.py:128]:
```suggestion
                --model_name_or_path hf-internal-testing/tiny-random-DistilBertForQuestionAnswering
```

**Review by echarlaix — APPROVED** [review]
Thanks a lot for your work @NoushNabi

**Comment by echarlaix** [comment]
To fix the code style check you can do the following : 
```
pip install .[quality]
make style
```

**Review by echarlaix — APPROVED** [review]
great work thanks @NoushNabi

**Pull request merged**